### PR TITLE
feat(typescript): worker mode for ts_project

### PR DESCRIPTION
**Pull request by mrmeku**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?
Issue Number: https://github.com/bazelbuild/rules_nodejs/issues/1726

## What is the new behavior?
Introduce experimental `supports_workers` property on ts_project to persist worker tsc --watch processes

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information
Unsupported on windows with current implementation which relies on suspending/resuming tsc --watch process

### CODE REVIEW

File: packages/typescript/internal/ts_project.bzl [hunk 1]
```diff
@@ -128,6 +143,7 @@ def _ts_project_impl(ctx):
             arguments = [arguments],
             outputs = outputs,
             executable = "tsc",
+            execution_requirements = execution_requirements,
```
Comment (alexeagle) [inline packages/typescript/internal/ts_project.bzl:146]: I would expect we need a mnemonic here to key the worker pool?

File: packages/typescript/internal/ts_project.bzl [hunk 2]
```diff
@@ -421,6 +443,35 @@ def ts_project_macro(
         )
         extra_deps.append("_validate_%s_options" % name)
 
+    is_windows = select({
+        "@bazel_tools//src/conditions:host_windows": True,
+        "//conditions:default": False,
+    })
+    supports_workers = supports_workers and not is_windows
```
Comment (alexeagle) [inline packages/typescript/internal/ts_project.bzl:450]: oops, this is checking if the `select` is false, not the result of it - so workers are totally turned off in this PR as-is. I'll push a fix...

**Comment by alexeagle** [comment]
@googlebot I consent.

File: packages/typescript/internal/ts_project.bzl (hunk 1 above)
Comment (mrmeku) [reply packages/typescript/internal/ts_project.bzl:171]: done

**Comment by alexeagle** [comment]
failed on RBE, feels like we shouldn't be trying to run worker code there?

```

(03:26:30) ERROR: /workdir/packages/typescript/test/ts_project/worker/BUILD.bazel:3:1: Compiling TypeScript project (worker mode) //packages/typescript/test/ts_project/worker:tsconfig [tsc -p packages/typescript/test/ts_project/worker/tsconfig_tsconfig.json] failed (Exit 1): tsconfig_worker.sh failed: error executing command
--
  | (cd /var/lib/buildkite-agent/.cache/bazel/_bazel_buildkite-agent/ec321eb2cc2d0f8f91b676b6d4c66c29/execroot/build_bazel_rules_nodejs && \
  | exec env - \
  | BAZEL_NODE_MODULES_ROOT='' \
  | COMPILATION_MODE=fastbuild \
  | bazel-out/k8-fastbuild/bin/packages/typescript/test/ts_project/worker/tsconfig_worker.sh @bazel-out/k8-fastbuild/bin/packages/typescript/test/ts_project/worker/big.js-0.params '--bazel_node_modules_manifest=bazel-out/k8-fastbuild/bin/packages/typescript/test/ts_project/worker/_tsconfig.module_mappings.json' --nobazel_patch_module_resolver)
  | Execution platform: @buildkite_config//config:platform
  | internal/modules/cjs/loader.js:797
  | throw err;
  | ^
  |  
  | Error: Cannot find module 'protobufjs'
  | Require stack:
  | - /b/f/w/bazel-out/k8-fastbuild/bin/packages/typescript/internal/worker/worker.js
  | - /b/f/w/bazel-out/k8-fastbuild/bin/packages/typescript/internal/worker/worker_adapter.js
  | at Function.Module._resolveFilename (internal/modules/cjs/loader.js:794:15)


```

**Comment by mistic** [comment]
@alexeagle why it should fail on RBE? 
@mrmeku it is also failing on windows and as I read across the code it looks like we are not going to support it. What happened with your exploration with PsSuspend? (https://www.npmjs.com/package/win-pause) I think it should work, the only thing is that we need to verify if pssuspend is installed. If it is we can proceed otherwise we can change support_workers to false

**Pull request merged**